mcs/class/Mono.Data.Tds/Mono.Data.Tds.Protocol/Tds70.cs

```diff
@@ -559,9 +559,8 @@ private void WriteParameterInfo (TdsMetaParameter param)
 			 * If the value is null, not setting the size to 0 will cause varchar
 			 * fields to get inserted as an empty string rather than an null.
 			 */
-			if (colType != TdsColumnType.IntN && colType != TdsColumnType.DateTimeN) {
```

Comment:
As far as I can see the if condition checks whether the column type is *not* a nullable Int/DateTime so it should work with the other types?